File: backend-repo/FaniverseBE/src/main/java/fantastic/faniverse/product/GeneralProduct/dto/GeneralProductRegisterRequest.java
package fantastic.faniverse.product.GeneralProduct.dto;

import fantastic.faniverse.product.dto.ProductRegisterRequest;
import fantastic.faniverse.product.GeneralProduct.domain.GeneralProductStatus;
import fantastic.faniverse.product.GeneralProduct.domain.GeneralProduct;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Valid
public class GeneralProductRegisterRequest extends ProductRegisterRequest {

    @NotEmpty(message = "가격 입력은 필수입니다.")
    private double price;

    private String imageUrl;

    public GeneralProduct toGeneralProductEntity(String imageUrl) {
        return GeneralProduct.builder()
                .title(getTitle())
                .category(getCategory())
                .content(getContent())
                .price(price)
                .generalProductStatus(GeneralProductStatus.SALE)
                .imageUrl(imageUrl)
                .build();
    }
}

File: frontend-repo/FaniverseFE2/app/src/main/java/com/example/login/model/GeneralProductRegisterRequest.kt
package com.example.login.model

import okhttp3.MultipartBody

data class GeneralProductRegisterRequest(
    val price: Double,
    override val title: String,
    override val category: String,
    override val content: String,
    override val imageUrl: String? = null,
    override val image: MultipartBody.Part?
) : ProductRegisterRequest(title, category, content, image, imageUrl)